src/main/java/io/confluent/connect/jdbc/source/JdbcSourceConnectorConfig.java

```diff
@@ -252,6 +258,31 @@ public boolean visible(String name, Map<String, Object> config) {
     }
   }
 
+  /**
+   * A recommender for table names that caches the table names for the last configuration.
+   */
+  private static class CachingTableRecommender extends TableRecommender {
+
+    private Map<String, Object> lastConfig;
+    private List<Object> results;
+
+    @Override
+    public List<Object> validValues(String name, Map<String, Object> config) {
+      if (lastConfig != null && lastConfig.equals(config)) {
```

Comment:
The `CachedTableRecommender` instance is per `ConfigDef` instance, and it seems like every validation attempt acquires a new `ConfigDef` instance. Is that right? If so, then each recommender instance should only be around for a short period of time and only to validate a single configuration.